packages/credentials/src/V1/KiltCredentialV1.ts

```diff
@@ -327,15 +328,76 @@ export function fromInput({
 
 const cachingCTypeLoader = newCachingCTypeLoader()
 
+// Helper function to check if CType is nested
+function cTypeTypeFinder(cType: ICType): boolean {
+  function hasRef(obj: any): boolean {
+    if (typeof obj !== 'object' || obj === null) return false
+    
+    if ('$ref' in obj) return true
+    
+    return Object.values(obj).some(value => {
+      if (Array.isArray(value)) {
+        return value.some(item => hasRef(item))
+      }
+      if (typeof value === 'object') {
+        return hasRef(value)
+      }
+      return false
+    })
+  }
+  return hasRef(cType.properties)
+}
+
+// Helper function to extract unique references from CType
+function extractUniqueReferences(cType: ICType): Set<string> {
```

Comment:
As above, change it.

Reply:
Here I can somewhat see the point of having an inner function; alternatively, you could solve it by taking an optional second argument:
 ```suggestion
function extractUniqueReferences(cType: ICType, references=new Set<string>()): Set<string> {
```

Reply:
Fixed at: https://github.com/KILTprotocol/sdk-js/pull/919/commits/77b9c3d89f213c51b6b428ade650cc04c50efc2b